app/antedecorators/gas.go

```diff
@@ -32,22 +35,112 @@ func GetGasMeterSetter(aclkeeper aclkeeper.Keeper) func(bool, sdk.Context, uint6
 }
 
 func getMessageMultiplierDenominator(ctx sdk.Context, msg sdk.Msg, aclKeeper aclkeeper.Keeper) uint64 {
+	// TODO: reason through whether it's reasonable to require non-* identifier for all operations
```

Comment:
Can you elaborate the diff between what you described and the current impl? also I think we want to discourage using * not just for type ALL but also for resource type like bank since that would make it blocking for most messages

Reply:
Yeah but having * for bank balances for example still allows things like staking messages to be unblocked. If the gas discount is the main incentive for parallelization, there should be SOME incentive to parallelize even if your contract needs to iterate over some addresses or something resulting in a * on a KV prefix otherwise people wont do anything aka blocking EVERYTHING, right?